packages/sdk/src/bot/client/types.test.ts

```diff
@@ -0,0 +1,291 @@
+import { MakeChannel, MakeIntegration } from '../../integration/generic'
+import { test } from 'vitest'
+import * as utils from '../../utils/type-utils'
+import { BaseBot, MakeBot } from '../generic'
+import * as types from './types'
+
+type FooBarBazIntegration = MakeIntegration<{
+  actions: {
+    doFoo: {
+      input: {
+        inputFoo: string
+      }
+      output: {
+        outputFoo: string
+      }
+    }
+    doBar: {
+      input: {
+        inputBar: number
+      }
+      output: {
+        outputBar: number
+      }
+    }
+    doBaz: {
+      input: {
+        inputBaz: boolean
+      }
+      output: {
+        outputBaz: boolean
+      }
+    }
+  }
+  events: {
+    onFoo: {
+      eventFoo: string
+    }
+    onBar: {
+      eventBar: number
+    }
+    onBaz: {
+      eventBaz: boolean
+    }
+  }
+  channels: {
+    channelFoo: MakeChannel<{
+      messages: {
+        messageFoo: {
+          foo: string
+        }
+      }
+    }>
+    channelBar: MakeChannel<{
+      messages: {
+        messageBar: {
+          bar: number
+        }
+      }
+    }>
+    channelBaz: MakeChannel<{
+      messages: {
+        messageBaz: {
+          baz: boolean
+        }
+      }
+    }>
+  }
+}>
+
+test('EnumerateActions should enumerate actions', () => {
```

Comment:
**question:** if we're not testing instances of the types, what is the benefit of using vitest, as opposed to just letting `tsc` throw errors?

Reply:
There are no real benefits really. Before that's exactly what I did, but I didn't want to pollute the source code with 300+ lines of tests. Also a file with extension `.test.ts` is colored differently by the IDE so it's clear that the file has a different purpose